x/sync/sync_test.go

```diff
@@ -26,39 +26,56 @@ import (
 	pb "github.com/ava-labs/avalanchego/proto/pb/sync"
 )
 
-var _ Client = (*mockClient)(nil)
-
 func newNoopTracer() trace.Tracer {
 	tracer, _ := trace.New(trace.Config{Enabled: false})
 	return tracer
 }
 
-type mockClient struct {
-	db DB
-}
+func newCallthroughSyncClient(ctrl *gomock.Controller, db merkledb.MerkleDB) *MockClient {
```

Comment:
Replaced `mockClient` with usage of the gomock mock since I don't think we need 2 mock implementations.